jvm/src/main/scala/io/glutenproject/expression/BinaryExpressionTransformer.scala

```diff
@@ -25,6 +25,27 @@ import io.glutenproject.substrait.expression.{ExpressionBuilder, ExpressionNode}
 import org.apache.spark.internal.Logging
 import org.apache.spark.sql.catalyst.expressions._
 
+class InstrTransformer(str: Expression, substr: Expression, original: Expression)
+  extends StringInstr(str: Expression, substr: Expression)
+    with ExpressionTransformer
+    with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    val strNode = str.asInstanceOf[ExpressionTransformer].doTransform(args)
+    val substrNode = substr.asInstanceOf[ExpressionTransformer].doTransform(args)
+    if (!strNode.isInstanceOf[ExpressionNode] || !substrNode.isInstanceOf[ExpressionNode]) {
+      throw new UnsupportedOperationException(s"Not supported yet.")
+    }
+
+    val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
+    val functionId = ExpressionBuilder.newScalarFunction(functionMap, ConverterUtils.makeFuncName(
+      ConverterUtils.INSTR, Seq(str.dataType, substr.dataType), FunctionConfig.OPT))
+    val expressionNodes = Lists.newArrayList(strNode, substrNode)
+    val typeNode = TypeBuilder.makeI32(original.nullable)
```

Comment:
Any other return type here? Maybe should not be hard-coded as i32.

Reply:
Because the return type of Velox function signature is i32.

Reply:
The return type should be aligned with Spark's expression. For this expression, Spark's return type is IntergerType so it is OK. For other cases, `getTypeNode` could be used to convert Spark datatype into Substrait type node.

Reply:
Thx